packages/shared/providers.yaml

```
@@ -3826,6 +3826,29 @@ ramp-sandbox:
         base_url: https://demo-api.ramp.com
     docs: https://docs.nango.dev/integrations/all/ramp
 
+rapidapi:
+    display_name: RapidAPI
+    categories:
+        - dev-tools
+    auth_mode: API_KEY
+    proxy:
+        headers:
+            x-rapidapi-key: ${apiKey}
+            x-rapidapi-host: ${connectionConfig.host}
+        base_url: https://${connectionConfig.host}
+    docs: https://docs.nango.dev/integrations/all/rapidapi
+    credentials:
+        apiKey:
+            type: string
+            title: API Key
+            description: The API key for your RapidAPI account
```

You can add pattern and example here 
```suggestion
            description: The API key for your RapidAPI account
            example: b7c156af2dmgh5c635305f3744bap168553jsp75193c8367ef
            pattern: '^[a-zA-Z0-9]+$'
```